exercises/exercise_2.md

```diff
@@ -19,9 +19,15 @@ Pytest will collect tests based on the following criteria:
 To run the tests, execute `pytest` in the directory containing your tests,
 or specify the path: `pytest <path_to_dir>`.
 
+Docs: https://docs.pytest.org/en/latest/getting-started.html#create-your-first-test
+
+### Asserts
 You can use `assert a == b` to check actual values against expected values.
 In general it's not needed to add an assert message, since Pytest does a great job
 reporting on failed tests. If you do want an assert message: `assert a == b, "a did not equal b"`
 or `assert a == b, "value of a: {} does not equal value of b: {}".format(a, b)`
 
-Docs: https://docs.pytest.org/en/latest/getting-started.html#create-your-first-test
\ No newline at end of file
+Pytest defaults to hard asserts, so a failing assert will fail the whole
+test without running the rest of it. See here for how to set up soft asserts:
+http://element34.ca/blog/soft-asserts-and-py-test.
```

Comment:
Haven't yet been able to make this work with test functions.
Also found a different approach: http://pythontesting.net/strategy/delayed-assert

Reply:
http://element34.ca/blog/soft-asserts-and-py-test uses `__multicall__` which is deprecated:
https://stackoverflow.com/questions/43009164/what-is-multicall-argument-and-why-it-is-deprecated

Reply:
New branch to figure out soft asserts: https://github.com/j19sch/building-an-api-testing-framework/tree/soft-assert